blockprod/src/detail/mod.rs

```diff
@@ -210,42 +214,30 @@ impl BlockProduction {
         Ok(consensus_data)
     }
 
+    #[allow(clippy::too_many_arguments)]
     fn generate_finalize_block_data(
         chain_config: &ChainConfig,
         chainstate_handle: &(dyn ChainstateInterface),
         best_block_index: &GenBlockIndex,
         block_height: BlockHeight,
+        current_timestamp: BlockTimestamp,
```

Comment:
Why can't we extract this from the best block index?

Reply:
I've cleaned up a few of the timestamp usages around blockprod, including proper naming i.e before, current didn't mean "now" sometimes.